polyfill/test/usertimezone.mjs

```diff
@@ -77,21 +77,29 @@ describe('Userland time zone', () => {
     });
   });
   describe('Trivial protocol implementation', () => {
-    const obj = {
+    class CustomTimeZone extends Temporal.TimeZone {
```

Comment:
I don't think this change is correct. We still want to test a plain-object implementation of the TimeZone protocol, and not an instance of TimeZone. I guess the "it converts to DateTime" test below fails, but that test should be removed because that's exactly the capability we're removing with this change.

Reply:
ah, OK, I see. I wasn't sure if we wanted to allow plain objects for `TimeZone` (there were some other tests that also use this pattern of introducing a fake subclass, and they have branding checks, too, so they misled me). I'll revert the commit.